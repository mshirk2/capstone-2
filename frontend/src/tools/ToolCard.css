.ToolCard {
    max-height: 500px;
    margin-bottom: 1rem;
    color: inherit;
    border-radius: 10px;
    border: none;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
}

.card-body {
    text-align: left;
}

img {
    width: 30%;
    float: left;
    margin-right: 1em;
  }

.link {
    color: black;
}