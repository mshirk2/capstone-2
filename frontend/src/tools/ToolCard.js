import React from "react";
import { Link } from "react-router-dom";
import placeholderImage from "../images/screwdriver-wrench-solid.svg";
import "./ToolCard.css"

function ToolCard({id, title, catalogCode, brand, model, condition, description, contents, tags, image}){

    if(!image) image = placeholderImage;

    return (
        <div className="ToolCard card">
            <div className="card-body">
                <Link to={`tools/${id}`} className="link">
                    {image && <img src={image} alt={title} />}
                    <h5 className="card-title">{title}</h5>
                </Link>
                <h6 className="card-subtitle mb-2 text-muted">{catalogCode}</h6>
                {brand && <div>Brand: {brand}</div>}
                {condition && <div>Condition: {condition}</div>}
                {tags && tags.map(tag => (<span key={tag} className="badge badge-light mr-2 mt-2">{tag}</span>))}
            </div>
        </div>
    )
}

export default ToolCard;