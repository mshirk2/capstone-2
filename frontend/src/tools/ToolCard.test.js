import React from "react";
import { render } from "@testing-library/react";
import ToolCard from "./ToolCard";
import { MemoryRouter } from "react-router";

it('renders without crashing', () =>{
    render(
        <MemoryRouter>
            <ToolCard />
        </MemoryRouter>
    )
});

it ("matches snapshot", () =>{
    const { asFragment } = render(
        <MemoryRouter>
            <ToolCard 
                title={"Testing tool"}
                catalogCode={33}
                condition={"good"}
            />
        </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
});