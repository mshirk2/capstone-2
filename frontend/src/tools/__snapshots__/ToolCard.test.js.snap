// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ToolCard card"
  >
    <div
      class="card-body"
    >
      <a
        class="link"
        href="/tools/undefined"
      >
        <img
          alt="Testing tool"
          src="screwdriver-wrench-solid.svg"
        />
        <h5
          class="card-title"
        >
          Testing tool
        </h5>
      </a>
      <h6
        class="card-subtitle mb-2 text-muted"
      >
        33
      </h6>
      <div>
        Condition: good
      </div>
    </div>
  </div>
</DocumentFragment>
`;
